subsys/bluetooth/services/hids.c

```diff
@@ -287,11 +287,13 @@ static ssize_t hids_outp_rep_read(struct bt_conn *conn,
 	ret_len = bt_gatt_attr_read(conn, attr, buf, len, offset, rep_data,
 				    rep->size);
 
-	if (rep->handler) {
-		struct bt_hids_rep report = {
-		    .data = buf,
-		    .size = rep->size,
-		};
+	struct bt_hids_rep report = {
```

Comment:
Consider extending `struct bt_hids_rep` with `uint8_t report_id` to keep the ID together with report data and to avoid changing the API. HID report ID is actually part of the HID report.

Reply:
that is a good idea, let's switch to it